test/integration/targets/vmware_portgroup/tasks/main.yml

```diff
@@ -0,0 +1,124 @@
+# Test code for the vmware_portgroup_facts module.
+# Copyright: (c) 2018, Abhijeet Kasurde <akasurde@redhat.com>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+- name: store the vcenter container ip
+  set_fact:
+    vcsim: "{{ lookup('env', 'vcenter_host') }}"
+
+- debug: var=vcsim
+
+- name: Wait for Flask controller to come up online
+  wait_for:
+    host: "{{ vcsim }}"
+    port: 5000
+    state: started
+
+- name: kill vcsim
+  uri:
+    url: http://{{ vcsim }}:5000/killall
+
+- name: start vcsim
+  uri:
+    url: http://{{ vcsim }}:5000/spawn?cluster=2
+  register: vcsim_instance
+
+- debug:
+    var: vcsim_instance
+
+- name: Wait for vcsim server to come up online
+  wait_for:
+    host: "{{ vcsim }}"
+    port: 443
+    state: started
```

Comment:
The preamble from the top of the file to here is pretty much the same for every test.  If not, it's darn close.  Think about factoring this away so someone could instead do:

```
- import_tasks: common/setup_vcsim.yml
  vars:
    clusters: 2
    hosts: 4
    vms: 20
```

Since I was asked to provide some simulator tests for open PRs I will probably do this eventually.

Reply:
This should be done globally for all other modules if @Akasurde thinks this makes sense. As far as I know, he's working on implementing a test environment with "real" ESXi hosts and vCenter servers and vcsim shouldn't be necessary anymore in the feature. 

I just extended the tests or copy'n'pasted the directory in all of my PRs. Therefore, your recommendation isn't that beneficial for me and I won't change it until it's officially required and then it should be changed for all modules in a PR... 

Those vcsim tests are more or less just a pain ** *** *** because a lot doesn't work anyway and my local tests in my environment are much more value than a simulator test...

Reply:
I am OK with `import_tasks` idea. We can do refactoring as part of separate PR where every single testcase is converted to suggested idea rather than doing it here. Let me know if it make sense.

Reply:
@jeking3 MagicMock is nice idea. I have seen that in one of the test case in your Pyvmomi patch.

Reply:
Ok. I will resolve this conversation. Please discuss it elsewhere. I'm happy to implement it if my PR didn't get approved before you defined it and if it's a standard for all modules.